inspire_disambiguation/core/helpers.py

```diff
@@ -320,6 +321,16 @@ def process_clustering_output(clusterer):
     return output
 
 
+def train_validation_split(curated_signatures_dict, split_fraction):
+    seed(99)
+    train_signatures_uuids = sample(list([*curated_signatures_dict]),
+                                    k=int(len([*curated_signatures_dict]) * split_fraction))
```

Comment:
This should be a `set` too for performance. I also have the impression you tried to randomly hit keys until this worked :)
```suggestion
    train_signatures_uuids = set(sample(curated_signatures_dict.keys()),
                                    k=int(len(curated_signatures_dict) * split_fraction)))
```